### DISCUSSION TITLE
Drawing a simple cube

### DISCUSSION THREAD

**Discussion by tiavision** [cetz-package/cetz#547] [Q&A] [answered]
When trying to draw a simple cube the edges seem to be to not match at all. I tried different approaches, but I just can't seem to get the edges to connect properly. Here is a simple example with a line:

```typst
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *
  
  line((0,0), (1,0), (1,1), (1,1,0.2), (1,0,0.2), (1,0), (0,0), (0,1), (1,1), (1,1,0.2), (0,1,0.2), (0,1))
})
```

Here is a screenshot of the output. Notice the edges on the top right in particular, but also on the bottom right and top left.

![Screenshot 2024-03-20 223400](https://github.com/johannes-wolf/cetz/assets/2929650/7ac80a58-d940-40bf-9386-865a7fd7a409)

**Comment by fenjalien** [comment, marked as answer]
You could try changing the stroke styling, specifically `join` (https://typst.app/docs/reference/visualize/stroke/#constructor-join) or `miter-limit` (https://typst.app/docs/reference/visualize/stroke/#constructor-miter-limit). With `set-style(stroke: (join: "bevel"))`: 
![image](https://github.com/johannes-wolf/cetz/assets/34489450/3d795b70-54c1-45bb-ac59-67c9c0fd1b19)
